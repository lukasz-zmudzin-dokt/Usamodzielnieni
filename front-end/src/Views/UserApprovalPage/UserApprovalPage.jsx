import React, { useState, useEffect, useContext} from "react";
import { UserContext } from "context";
import {Accordion, Alert, Card, Container} from "react-bootstrap";
import UserToApprove from "./components/UserToApprove";
import { getUsersToApprove } from "./functions/apiCalls";

const UserApprovalPage = () => {

    const context = useContext(UserContext);
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState(false);
    const [users, setUsers] = useState([]);

    useEffect(() => {
        const loadUsers = async(token, setUsers) => {
            setLoading(true);
            try {
                let res = await getUsersToApprove(token);
                setUsers(res);
                setLoading(false);
            } catch (e) {
                setError(true);
                setLoading(false);
            }
        };
        loadUsers(context.token, setUsers);
    }, [context.token]);

    const message = loading ? (
        <Alert variant="info" className="m-3">Ładuję...</Alert>
    ) : error ? (
        <Alert variant="danger" className="m-3">Ups, wystąpił błąd.</Alert>
    ) : users.length === 0 ? (
        <Alert variant="info" className="m-3">Brak kont do zatwierdzenia</Alert>
    ) : null;

    return (
        <Container>
            <Accordion>
                <Card>
                    <Card.Header as={"h2"}>
                        Konta do zatwierdzenia
                    </Card.Header>
                    <Card.Body className="p-0">
                        { message ? message : null }
                        { users.map((user) => <UserToApprove user={user} key={user.id} />) }
                    </Card.Body>
                </Card>
            </Accordion>
        </Container>
    );
};

export default UserApprovalPage;