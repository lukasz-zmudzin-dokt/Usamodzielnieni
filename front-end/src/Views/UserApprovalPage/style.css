.background {
    background-image: url("../../assets/menu-backdrop.png");
    background-size: contain;
    background-repeat: no-repeat;
    background-position: center;
    position: relative;

    height: 100vh;
    flex-direction: column;
}

.users-approval-card {
    padding-top: 10vh;
}
.no-border {
    border: none !important;
}
.border {
    border: gray;
}