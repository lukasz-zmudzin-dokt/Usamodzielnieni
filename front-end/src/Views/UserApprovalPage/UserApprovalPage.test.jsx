import React from "react";
import { render, waitForElement, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import UserApprovalPage from "./UserApprovalPage";

describe("UserApproval", () => {
    let failFetch;
    let apiUsers = [
        {
            date_joined: "2020-04-25T23:31:59.239639+02:00",
            email: "qwe@qwe.vcb",
            id: "fcd32247-90ac-4394-92ca-27252bdc2ee9",
            last_login: "2020-04-25T23:31:59.239631+02:00",
            status: "Waiting for verification",
            type: "Standard",
            username: "testowyuser123"
        },
        {
            date_joined: "2019-04-25T23:31:59.239639+02:00",
            email: "qwe@qwe.vcb2",
            id: "fcd32247-90ac-4394-92ca-27252bdc2ee0",
            last_login: "2020-04-25T23:32:59.239631+02:00",
            status: "Waiting for verification",
            type: "Employer",
            username: "testowyuser1232"
        }
    ];

    beforeAll(() => {
        global.fetch = jest.fn().mockImplementation((input, init) => {
            return new Promise((resolve, reject) => {
                if(failFetch) {
                    resolve({ status: 500 });
                } else {
                    resolve({ status: 200, json: () => Promise.resolve(apiUsers) });
                }
            });
        });
    });

    beforeEach(() => {
        failFetch = false;
        jest.clearAllMocks();
    });

    it("should match snapshot", async () => {
        const { container, getByText } = render (
            <MemoryRouter>
                <UserApprovalPage />
            </MemoryRouter>
        );
        await waitForElement(() => getByText("testowyuser123 (Standard)"));
        expect(container).toMatchSnapshot();
    });

    it("should load users", async () => {
        const { getByText } = render (
            <MemoryRouter>
                <UserApprovalPage />
            </MemoryRouter>
        );
        await waitForElement(() => getByText("testowyuser123 (Standard)"));
        expect(getByText("testowyuser123 (Standard)")).toBeInTheDocument();
    });

    it("should view alert at api fail", async () => {
        failFetch = true;
        const { getByText } = render (
            <MemoryRouter>
                <UserApprovalPage />
            </MemoryRouter>
        );

        await waitForElement(() => getByText("Ups, wystąpił błąd."));
        expect(getByText("Ups, wystąpił błąd.")).toBeInTheDocument();
    });

    it("should view alert at api returning no users", async () => {
        apiUsers = [];
        const { getByText } = render (
            <MemoryRouter>
                <UserApprovalPage />
            </MemoryRouter>
        );
        await waitForElement(() => getByText("Brak kont do zatwierdzenia"));
        expect(getByText("Brak kont do zatwierdzenia")).toBeInTheDocument();
    });
});