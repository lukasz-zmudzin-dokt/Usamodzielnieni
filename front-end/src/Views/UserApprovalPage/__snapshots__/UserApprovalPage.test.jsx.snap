// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserApproval should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="card"
    >
      <h2
        class="card-header"
      >
        Konta do zatwierdzenia
      </h2>
      <div
        class="p-0 card-body"
      >
        <div
          class="accordion"
        >
          <div
            class="border-left-0 border-right-0 border-bottom-0 card"
          >
            <div
              class="card-header"
            >
              testowyuser123
               (
              Standard
              )
            </div>
            <div
              class="collapse"
            >
              <div
                class="card-body"
              >
                <div
                  class="list-group list-group-flush"
                >
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="detailsItem col-md-4"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Nazwa użytkownika
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Imię
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Nazwisko
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Email
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Telefon
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Adres
                        </div>
                        <div
                          class="detailsItem__value"
                        >
                          <p />
                          <p>
                             
                          </p>
                          <p>
                             
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="justify-content-center row"
                    >
                      <button
                        class="btn btn-success"
                        type="button"
                      >
                        Akceptuj
                      </button>
                      <button
                        class="ml-3 btn btn-danger"
                        type="button"
                      >
                        Odrzuć
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="border-left-0 border-right-0 border-bottom-0 card"
          >
            <div
              class="card-header"
            >
              testowyuser1232
               (
              Employer
              )
            </div>
            <div
              class="collapse"
            >
              <div
                class="card-body"
              >
                <div
                  class="list-group list-group-flush"
                >
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="detailsItem col-md-4"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Nazwa użytkownika
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Imię
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Nazwisko
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Email
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Telefon
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="row"
                    >
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          Adres
                        </div>
                        <div
                          class="detailsItem__value"
                        >
                          <p />
                          <p>
                             
                          </p>
                          <p>
                             
                          </p>
                        </div>
                      </div>
                      <div
                        class="detailsItem col"
                      >
                        <div
                          class="detailsItem__label"
                        >
                          NIP
                        </div>
                        <div
                          class="detailsItem__value"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="justify-content-center row"
                    >
                      <button
                        class="btn btn-success"
                        type="button"
                      >
                        Akceptuj
                      </button>
                      <button
                        class="ml-3 btn btn-danger"
                        type="button"
                      >
                        Odrzuć
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
